jetty-core/jetty-http2/jetty-http2-client-transport/src/main/java/org/eclipse/jetty/http2/client/transport/internal/HttpReceiverOverHTTP2.java

```diff
@@ -253,15 +269,12 @@ public Runnable onTimeout(TimeoutException failure, Promise<Boolean> promise)
             promise.succeeded(false);
             return null;
         }
-        return new Invocable.ReadyTask(Invocable.InvocationType.NON_BLOCKING, () ->
-            promise.completeWith(exchange.getRequest().abort(failure))
-        );
+        return () -> promise.completeWith(exchange.getRequest().abort(failure));
```

Comment:
Again due to `onTimeout()`'s returned runnable is executed immediately? This should be on `onTimeout()`'s javadoc.